### ISSUE TITLE
[Lua request] Get BizHawk version with client.getversion

### ISSUE THREAD

**Issue by brunovalads** [TASVideos/BizHawk#1271]
In order to make my scripts more accessible, would be interesting a function like `client.getversion` (not sure if `client` is the best class, seems to be) that returns the BizHawk version as a string (this information is stored somewhere), and it can be used to forbid features in the scripts that use lua functions developed later (e.g. a lot of lua functions doesn't exist in 1.X). This is interesting to avoid making multiple versions of the same script.

**Comment by brunovalads** [comment]
Oh nice! 

> Ultimately though, you have no way of differentiating between v1.x.x clients and v2 clients up to 2.3.0 (as none of these will have the new Lua method).

Yea that's exactly what I thought when I was reading your reply, this feature will only be really useful in the future when older BizHawk versions are completely forgotten. So for now, the way I'm used to identify the version is the way Amaraticando did in the SMW script:
```lua
biz.features = {
  backcolor_default_arg = (emu.setislagged == nil), -- < 1.11.5
  gui_text_backcolor = (gui.DrawFinish == nil), -- < 1.11.7
  support_extra_padding = (client.SetGameExtraPadding ~= nil) and (gui.DrawFinish ~= nil), -- 1.11.7
}
```
So in the future, I would need to make the following:
```lua
local bizhawk_version

if client.getversion ~= nil then -- is "2.4.0" or higher
  bizhawk_version = client.getversion()
else -- 2.3.X or lower
  if emu.getregisters().H ~= nil then bizhawk_version = "2.3.0"
  elseif emu.getluacore ~= nil then bizhawk_version = "2.2.2"
  elseif memory.hash_region ~= nil then bizhawk_version = "2.2.1"
  elseif tastudio.getbranches ~= nil then bizhawk_version = "2.2.0"
  --elseif  ~= nil then bizhawk_version = "2.1.1"
  elseif forms.pictureBox ~= nil then bizhawk_version = "2.1.0"
  elseif SQL ~= nil then bizhawk_version = "2.0.1"
  elseif LuaCanvas.defaultForeground ~= nil then bizhawk_version = "2.0.0"
end
```
and make a few more conditionals to detect the 1.X versions as well.

**Comment by vadosnaprimer** [comment]
Yeah even if we add this now, all the previous releases would have to use hacks like this.

**Comment by Asnivor** [comment]
@vadosnaprimer presumably we could hardcode these hacks on the c# side, and return a version specifically for Lua.

The question is...should we..

**Comment by Asnivor** [comment]
actually, ignore me. That wouldnt work for older versions :/

**Comment by brunovalads** [comment]
Well, implement it now (if it's simple) and in 1 year it will be used without problems, because people wouldn't dare to use older versions (the only reasonable argument for using older versions is to finish a TAS you started there)

**Comment by Asnivor** [comment]
@vadosnaprimer can you think of any reason why we shouldnt be manually keeping this up to date in master? 

https://github.com/TASVideos/BizHawk/blob/49d067a665a4eab9363e0bdc7e261ed5e20a17c3/Version/VersionInfo.cs#L5

It makes no difference for release as this is generated at release time (as far as I can tell). But it would help from the point of view of getting the actual version from a dev build...

**Comment by vadosnaprimer** [comment]
For old versions one can simply host a manually created empty file with the emu version as its name, lua checking this file's existence. Or a file called version.txt, lua reading the version from it.

**Comment by vadosnaprimer** [comment]
@Asnivor since you probably have this set up already, just make a PR, and then we'll deal with manual version assigning.

**Comment by Asnivor** [comment]
@vadosnaprimer 

To be clear, we are going to keep this updated going forward? : https://github.com/TASVideos/BizHawk/commit/49d067a665a4eab9363e0bdc7e261ed5e20a17c3

And Lua will handle checking for a version txt file if the version <= 2.3 ?

If so, I will push a PR soon (as yes, I have a local branch somewhere with client.getversion in place)
